### ISSUE TITLE
Cannot parse input literal

### ISSUE THREAD

**Issue by micimize** [comigor/artemis#43]
The following mutation fails with due to an `{}` input literal with
`Error on line 2, column 43 of lib/gql/raw/SignIn.gql: Expected an object field name`. Generation succeeds if the literal is converted into an input `$variable`

```graphql
mutation SignIn {
    authenticated: currentAppUser(input: {}) {
        user: appUser {
            __typename
            entityId
        }
    }
}
```

```yaml
dependencies:
  artemis: ^2.0.4
  equatable: ^0.6.1
  gql: ^0.10.0
```

<details>
<summary>
full <code>flutter pub run build_runner build --verbose</code> output
</summary>

```
[INFO] Generating build script...
[INFO] Generating build script completed, took 318ms

[INFO] BuildDefinition:Initializing inputs
[INFO] BuildDefinition:Reading cached asset graph...
[INFO] BuildDefinition:Reading cached asset graph completed, took 120ms

[INFO] BuildDefinition:Checking for updates since last build...
[INFO] BuildDefinition:Checking for updates since last build completed, took 1.0s

[INFO] Build:Running build...
[SEVERE] artemis:artemis on lib/$lib$:

Error on line 2, column 43 of lib/gql/raw/SignIn.gql: Expected an object field name
  ╷
2 │     authenticated: currentAppUser(input: {}) {
  │                                           ^
  ╵
package:gql/src/language/parser.dart 61:5    _Parser._expectToken
package:gql/src/language/parser.dart 553:19  _Parser._parseName
package:gql/src/language/parser.dart 324:18  _Parser._parseObjectField
package:gql/src/language/parser.dart 321:7   _Parser._parseNonConstObjectField
package:gql/src/language/parser.dart 152:22  _Parser._parseAny
package:gql/src/language/parser.dart 409:17  _Parser._parseObject
package:gql/src/language/parser.dart 341:16  _Parser._parseValue
package:gql/src/language/parser.dart 451:14  _Parser._parseArgument
package:gql/src/language/parser.dart 442:44  _Parser._parseNonConstArgument
package:gql/src/language/parser.dart 117:22  _Parser._parseMany
package:gql/src/language/parser.dart 130:14  _Parser._maybeParseMany
package:gql/src/language/parser.dart 434:57  _Parser._parseArguments
package:gql/src/language/parser.dart 535:23  _Parser._parseField
package:gql/src/language/parser.dart 487:12  _Parser._parseSelection
package:gql/src/language/parser.dart 117:22  _Parser._parseMany
package:gql/src/language/parser.dart 474:21  _Parser._parseSelectionSet
package:gql/src/language/parser.dart 252:21  _Parser._parseOperationDefinition
package:gql/src/language/parser.dart 204:18  _Parser._parseExecutableDefinition
package:gql/src/language/parser.dart 173:18  _Parser._parseDefinition
package:gql/src/language/parser.dart 117:22  _Parser._parseMany
package:gql/src/language/parser.dart 159:22  _Parser._parseDocument
package:gql/src/language/parser.dart 41:27   _Parser.parse
package:gql/src/language/parser.dart 17:17   parse
package:gql/src/language/parser.dart 27:5    parseString
package:artemis/builder.dart 68:30           GraphQLQueryBuilder.build.<fn>

[INFO] Build:Running build completed, took 162ms

[INFO] Build:Caching finalized dependency graph...
[INFO] Build:Caching finalized dependency graph completed, took 74ms

[SEVERE] Build:
Failed after 246ms
pub finished with exit code 1

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      pubInteractively (package:flutter_tools/src/dart/pub.dart:244:5)
<asynchronous suspension>
#2      PackagesForwardCommand.runCommand (package:flutter_tools/src/commands/packages.dart:239:11)
<asynchronous suspension>
#3      FlutterCommand.verifyThenRunCommand
(package:flutter_tools/src/runner/flutter_command.dart:557:18)
#4      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#5      _rootRunUnary (dart:async/zone.dart:1132:38)
#6      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#7      _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#8      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#9      Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#10     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#11     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552:7)
#12     _rootRun (dart:async/zone.dart:1124:13)
#13     _CustomZone.run (dart:async/zone.dart:1021:19)
#14     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#15     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#16     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#17     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#18     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#19     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```
</details>

**Comment by klavs** [comment]
That's on `package:gql`. I guess the parser currently expects at least one field in the input object. Let me check the spec and the reference implementation.

**Comment by micimize** [comment]
@klavs Looks like it's just empty literals. I made it `{ id: "" }` and it worked

Default values like this fail on `]`:
```graphql
mutation SomeMutation(
  $someInput: SomeInput = { some: [] }
) {
  someMutation(input: $someInput) {
       success
    }
  }
}
```

The actual error:
```
Error on line 9, column 77 of lib/gql/raw/UpsertEventRecord.gql: Unexpected token found when parsing a value
  ╷
9 │   $metricRecordBatchInput: UpsertMetricRecordBatchInput = { metricRecords: [] }
  │                                                                             ^
  ╵
```

**Comment by klavs** [comment]
@micimize yes, I found the cause – 2 wrong rows and 1 unnecessary function. It affects empty objects and empty lists. I'll be able to fix it after 3h.

**Comment by klavs** [comment]
@micimize fixed and published in `gql` v0.11.1

Unfortunately, Artemis is still using `gql` v0.10.0 so we still need to update Artemis

**Comment by comigor** [comment]
Cool, let's update it!
